library/manageiq_alert.py

```diff
@@ -0,0 +1,282 @@
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password), verify_ssl=miq_verify_ssl, ca_bundle_path=ca_bundle_path)
+        self.changed       = False
+
+    def find_alert_by_description(self, description):
+        """ Searches the alert description in ManageIQ.
+
+        Returns:
+            the alert id if it exists in manageiq, None otherwise.
+        """
+        try:
+            response = self.client.get('{api_url}/alert_definitions?expand=resources'.format(api_url=self.api_url))
+        except Exception as e:
+            self.module.fail_json(msg="Failed to query alerts: {error}".format(error=e))
+        alerts = response.get('resources', [])
+        return next((alert['id'] for alert in alerts if alert['description'] == description), None)
+
+    def delete_alert(self, description):
+        """Deletes the alert from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        alert_id = self.find_alert_by_description(description)
+        if not alert_id:  # alert doesn't exist
+            return dict(
+                changed=self.changed,
+                msg="Alert {description} does not exist in manageiq".format(description=description))
+        try:
+            url = '{api_url}/alert_definitions/{alert_id}'.format(api_url=self.api_url, alert_id=alert_id)
+            result = self.client.post(url, action='delete')
+        except Exception as e:
+            self.module.fail_json(msg="Failed to delete alert {description}: {error}".format(description=description, error=e))
+        self.changed = True
+        return dict(changed=self.changed, msg=result['message'])
+
+    def alert_update_required(self, alert_id, description, expression, miq_entity, options, enabled):
+        """ Returns true if the expression, miq_entity, options, or enabled passed for
+            the alert differ from the alert's existing ones, False otherwise.
+        """
+        url = "{api_url}/alert_definitions/{alert_id}".format(api_url=self.api_url, alert_id=alert_id)
+        try:
+            result = self.client.get(url)
+        except Exception as e:
+            self.module.fail_json(msg="Failed to get alert {description} details. Error: {error}".format(description=description, error=e))
+
+        # remove None values from expression and options dicts
+        current_expression = dict((k, v) for k, v in result['expression']['exp'].items() if v is not None)
+        current_options = dict((k, v) for k, v in result['options'].items() if v is not None)
+
+        attributes_tuples = [(current_expression, expression), (result['db'], miq_entity), (current_options, options), (result['enabled'], enabled)]
+        for tup in attributes_tuples:
+            if tup[1] is not None and tup[0] != tup[1]:
```

Comment:
OK, I see update/create have trivial logic - just a dict {'foo': foo,...}.
Still, consider  uilding that dict early (in main?) and passing it around.  That way functions will have less arguments, and comparisons will be 1:1 eg. `current['db']` vs `desired ['db']`